app/src/main/java/com/willowtree/vocable/presets/CategoriesFragment.kt

```diff
@@ -58,19 +55,16 @@ class CategoriesFragment : BaseFragment<CategoriesFragmentBinding>() {
             val categoryButton =
                 CategoryButtonBinding.inflate(inflater, binding.categoryButtonContainer, false)
             with(categoryButton.root) {
-                tag = category
-                action = {
-                    viewModel.onCategorySelected(category.categoryId)
-
-                }
-                text = localizedResourceUtility.getTextFromCategory(category)
+                setUpDisplayAndAction(
+                    category = category
+                ) { viewModel::onCategorySelected }
```

Comment:
I'm not 100% certain that `viewModel::onCategorySelected` gets called when the lambda gets invoked here- double-check for me?

I think `{ viewMode.onCategorySelected() }` would do it for sure, uncertain with the `::` syntax

Reply:
I think your instinct is correct. It would need to be passed as a parameter, not in a lambda body.

Reply:
Looks like it's getting called whenever I switch categories. I think that when CategoryButton sets up its action with this funciton passed in (line 40), it makes the actual call. Behavior (user-facing and with breakpoints) seems to match main, which is the goal

Reply:
Hmmm, I'm possibly not looking at the right thing here, since tapping the button doesn't seem to work. I'll revert to passing it the other way - unfortunately this means even less gets refactored into the button itself